ROOTFS/etc/waggle/sanity/fatal/modem0_status.test

```diff
@@ -1,5 +1,47 @@
 #!/bin/bash
 
-echo "Modem0 Status Test: Begin"
-# use connectivity helper function
-/etc/waggle/sanity/_helper/connectivity_status.test modem0
+testname="Modem0 Status Test"
+
+pass() { echo "$testname: $1 PASS"; exit 0; }
+fail() { echo "$testname: $2 FAIL"; exit "$1"; }
+
+echo "$testname: Begin"
+
+# check system manifest to see if this test needs to be done.
+if ! manifest=$(jq .connectivity.modem0.present /etc/waggle/node_manifest.json); then
+    fail 1 "could not read manifest"
+fi
+
+if echo "$manifest" | grep -q false; then
+    pass "modem not required by manifest"
+fi
+
+# if already has an ip address, then we know we're good.
+if ip addr show dev modem0 2>/dev/null | grep -q "inet .*/.*"; then
+    pass "modem has ip address"
+fi
+
+# get first modem detected
+if ! modem=$(mmcli -L | grep -m1 -o '/org/freedesktop/ModemManager1/Modem/[0-9]*'); then
```

Comment:
this is probably fine.. just kinda curious if this path could ever be slightly different?